DOCSP-19966: fix pojo constructor and annotation description (#197)
mongodb/docs-java: source/fundamentals/data-formats/pojo-customization.txt

   * - ``BsonCreator``
     - Marks a public constructor or a public static method as the creator for

>        new instances of the class. All parameters in the constructor must
>        be annotated with a ``BsonProperty``. You must use the annotation

Reviewer: Issue: I think this contains passive voice, consider revising.

```suggestion
       new instances of the class. All parameters in the constructor must
       have a ``BsonProperty`` annotation. You must use the annotation
```

Author: I'll reword to be more specific, not sure the suggestion provides enough direction for the reader.

       and/or deserialize a property.

   * - ``BsonProperty``

>      - Specifies a custom document field name when converting the POJO

Reviewer: Question: Curious why "key" was changed to "field name". Any reason?

Author: This is for clarification. It matches the terminology we use in the rest of our MongoDB documentation. In MongoDB documents, we call them field names rather than keys (more applicable to describing JSON).

>        field to BSON. You can include a discriminator if you need one for
>        serializing nested POJOs.

Reviewer: Issue: Consider moving the "if" statement to the [beginning of the sentence](https://docs.mongodb.com/meta/style-guide/writing/write-clear-concise-sentences-paragraphs/#use-a-consistent-sentence-structure). 

Comment: This sentence was tough for me to understand. I think changing this from "if you need one for serializing..." to "If you must serialize a nested POJO" could help clarify the use case.

```suggestion
       field to BSON. If you must serialize a field containing a nested POJO, you can include a discriminator to
       specify how this field is to be serialized.
```

Author: I'm not sure I see consistent sentence structure in this suggestion -- none of the settings here are using "If... then" as descriptions and don't think that would be an appropriate way to describe them.

Instead, I'll make it consistent with the structure of the previous sentence. "Optionally specifies the value of a discriminator..."

Author: Another thought -- most settings/method descriptions contain the following info:
- description of what its effects are
- optional: describes default values and behaviors of other values
- optional: examples
- optional: direction on where to go for more information

I don't think consistent sentence structure across all these types of info is useful, but we could try to stick to the same form for each individual type of information.

In these sentences, it's describing what its effects are (both required and optional values), so keeping the same structure seems like a good idea.

Author: Updated sentence to remove "if you need one" which was perhaps the point of confusion.
